numerics/double_precision_body.hpp

```diff
@@ -14,15 +16,45 @@ namespace internal_double_precision {
 
 using geometry::PointOrMultivectorSerializer;
 using geometry::QuantityOrMultivectorSerializer;
+using quantities::Abs;
+using quantities::Quantity;
+using quantities::SIUnit;
+
+// Assumes that |T| and |U| have a memory representation that is a sequence of
+// |double|s, and returns the conjunction of componentwise
+// |left[i]| >= |right[i]| or left[i] == 0.
+template<typename T, typename U>
+bool ComponentwiseGreaterThanOrEqualOrZero(T const& left, U const& right) {
+  static_assert(sizeof(left) == sizeof(right),
+                "Comparing types of different sizes");
+  static_assert(sizeof(left) % sizeof(double) == 0,
+                "Types are not sequences of doubles");
+  constexpr int size = sizeof(left) / sizeof(double);
+  std::array<double, size> left_doubles;
+  std::memcpy(left_doubles.data(), &left, sizeof(left));
+  std::array<double, size> right_doubles;
+  std::memcpy(right_doubles.data(), &right, sizeof(right));
+  bool result = true;
+  for (int i = 0; i < size; ++i) {
+    result &=
+        (Abs(left_doubles[i]) >= Abs(right_doubles[i]) || left_doubles[i] == 0);
+  }
+  return result;
+}
+
+template<typename T, typename U>
+bool ComponentwiseGreaterThanOrEqualOrZero(DoublePrecision<T> const& left,
+                                           DoublePrecision<U> const& right) {
```

Comment:
Why do we need this function?  Technically it's incorrect.  I wonder if we could move it to the test, given that everything is in a single file anyway.